launcher/jvm_tooling.cpp

```diff
@@ -50,6 +57,32 @@ constexpr auto kJazzerBazelRunfilesPath =
     "jazzer_standalone_deploy.jar";
 constexpr auto kJazzerFileName = "jazzer_standalone.jar";
 
+// Returns the absolute path to the current executable. Compared to argv[0],
+// this path can always be used to locate the Jazzer JAR next to it, even when
+// Jazzer is executed from PATH.
+std::string getExecutablePath() {
+#if defined(__APPLE__)
+  char buf[655536];
+  uint32_t buf_size = sizeof(buf);
+  if (!_NSGetExecutablePath(buf, &buf_size)) {
+    return "";
+  }
+  return {buf};
```

Comment:
I'm not a C++ expert, but isn't this returning stack memory to the caller?
I.e. the caller will be using memory after it has been "freed"?

Reply:
Thanks for taking a look. Note the curly braces, which end up passing `buf` to constructor (5) at https://en.cppreference.com/w/cpp/string/basic_string/basic_string - that constructor copies the null-terminated string.